feature-libs/user/profile/components/update-email/update-email-component.service.spec.ts

```diff
@@ -147,6 +159,18 @@ describe('UpdateEmailComponentService', () => {
         service.save();
         expect(service.form.reset).toHaveBeenCalled();
       });
+
+      it('should set the redirect url to the home page before navigating to the login page', () => {
+        service.save();
+        expect(authRedirectService.setRedirectUrl).toHaveBeenCalledWith(
+          routingService.getUrl({ cxRoute: 'home' })
+        );
+        authService.coreLogout().then(() => {
+          expect(authRedirectService.setRedirectUrl).toHaveBeenCalledBefore(
+            routingService.go
+          );
+        });
+      });
```

Comment:
The test finishes immediately, even before the `expect` in the promise's callback is executed (line 169).

Let's wrap this whole test function in `waitForAsync(...)` so Jasmine knows to wait with completing the test, until async code is done. Btw. There are more test with similar issue in this file. It would be nice to also add `waitForAsync(...)` there.

---
Notes:
There are a couple ways to handle async tests:
- [using native Jasmine's `done()` function](https://angular.io/guide/testing-components-scenarios#jasmine-done)
- [**using Angular test wrapper `waitForAsync` (my personal preference over `done()`**)](https://angular.io/guide/testing-components-scenarios#async-test-with-waitforasync)
- [using Angular test wrapper `fakeAsync()` + `tick()` (only for testing sophiticated async flows; not needed in our case)](https://angular.io/guide/testing-components-scenarios#async-test-with-fakeasync)